docs/building/forms.md

```diff
@@ -0,0 +1,107 @@
+A form is like a collection of menus or raw responses and it is basically collecting information from the user.
+
+It is composed by a **header**, a **body** and a **footer**. The body represents form items and these can be slightly modified menus or raw responses.
+
+The form items are being processed in a successive manner. After all the form items have been processed, the user can confirm his choices or reset any of them.
+
+## Json Structure
+
+Your server must return a JSON response with a structure similar to the one below:
+
+```
+{'body': [{'type': 'string',
+           'name': 'descr',
+           'description': 'Provide a description',
+           'method': 'POST',
+           'path': '/callback-path'},
```

Comment:
Here is a question regarding the wizard flow:
So in this wizard step a path is defined. Does this mean that after the user insert the input for this wizard step, Nautilus will make a request with the input at that path?

Reply:
Yes, it will trigger a request if the path is set. Method defaults to get if not mentioned otherwise. It is step post processing

Reply:
Like ?name=choice in the post or get

Reply:
Ok. And how much time does Nautilus wait to receive the response from the app before sending the next step to the user? Or is it an async request for which Nautilus doesn't wait?

Reply:
That's a good question and honestly this is not implemented in nautilus yet. It has to be async, it's not like the on wizard finished hook, where a response is needed.

This post processing will be bundled with step validation in the next version.

Reply:
I'm not sure you saw the edit from the first message in this thread.

I suggest to not implement this in Nautilus and remove `method` and `path` from schema. If we support this feature we will have other issues to cover (eg. what happens when the user cancels the wizard after a partial data has been submited to the app?)

Reply:
Sounds good to me. The validation still needs to be implemented though